woocommerce/woocommerce-ios
# [Mobile Payments] Use reference instead of name when retrieving IPP Plugins

## Description

With this PR we change the way we retrieve the Card Present Payments Plugins when performing the onboarding checks. Before we checked the plugin name, but after a problem we had when they changed the in WCPay beta, we decided that it's safer to check the plugin reference, e.g., `"woocommerce-payments/woocommerce-payments",` which is less likely to change. WIth this we also are consistent with Android, which uses the same approach.

EDIT: As the folder can also change, we only evaluate now the filename without extension: pdfdoF-3pF-p2#comment-4619

Slack Convo: p1689676894933749-slack-C025A8VV728 

## Known Caveats
This solution brings some suboptimal parts with it. Before we could directly query for the name to the database, but now, as we have to remove the file extension, we have to bring all plugins to memory and then filter them. Feel free to add your opinion about it.

## Testing instructions


Check all possible combinations for IPP plugins, it should behave as before:

- WCPay and Stripe both installed and active. The IPP Onboarding should prompt you to select one of them.

<img src="https://github.com/woocommerce/woocommerce-ios/assets/1864060/011c6ffd-d997-4989-83ff-fa10f66d9403" width="375">

- WCPay Plugin is only installed and active. The IPP Onboarding should success and you should be able to collect a payment.
- Stripe Plugin is only installed and active. The IPP Onboarding should success and you should be able to collect a payment.
- No plugins active. The IPP Onboarding should fail and ask you to activate WCPay.
<img src="https://github.com/woocommerce/woocommerce-ios/assets/1864060/bc421834-ee92-4d77-8a20-4e1ab381cccf" width="375">

## Screenshots

See above

---
